src/Microsoft.AspNet.Razor.Runtime/TagHelpers/ITypeInfo.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Reflection;
+
+namespace Microsoft.AspNet.Razor.Runtime.TagHelpers
+{
+    /// <summary>
+    /// Contains type metadata.
+    /// </summary>
+    public interface ITypeInfo : IMemberInfo
+    {
+        /// <summary>
+        /// Fully qualified name of the type.
+        /// </summary>
+        string FullName { get; }
+
+        /// <summary>
+        /// Gets <see cref="IPropertyInfo"/>s for all properties of the current type excluding indexers.
```

Comment:
Though "indexer" has a well-defined meaning here, we confuse things a bit elsewhere in the `TagHelper` world. For example `TagHelperAttributeDescriptor.IsIndexer` is about how the property is treated when used as an attribute in Razor files. Perhaps useful to distinguish this case more completely.